spec/parsers/date_parser_spec.rb

```diff
@@ -0,0 +1,18 @@
+require "spec_helper"
+
+describe DateParser do
+  subject(:date_parser) { described_class.new(date_param) }
+
+  let(:date_param) { "I act like it's my birthday, every day" }
+  let(:date_string_parser) { instance_double(DateStringParser, parse: Date.new(1989, 12, 13)) }
+
+  before do
+    allow(DateStringParser).to receive(:new).and_return(date_string_parser)
```

Comment:
I would have thought we don't need to mock this and use doubles and instead can call original and assert it was called?

Reply:
Fair – the simplest thing is probably just to accept the coupling altogether and test the return value, then we don't even need to assert. Given how strong a dependency there is between them all I doubt it's too much of an issue!